sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -1,112 +1,178 @@
-import com.horizen.proposition.Proposition
-import com.horizen.serialization.Views
 import com.horizen.transaction.Transaction
 import org.web3j.crypto.TransactionDecoder
-import org.web3j.crypto.transaction.`type`.TransactionType
-import org.web3j.protocol.core.methods.response.{EthBlock, Log, TransactionReceipt, Transaction => Web3jTransaction}
 import org.web3j.utils.Numeric
 import scorex.core.NodeViewHolder.CurrentView
-import scorex.crypto.hash.Keccak256
 
+import java.math.BigInteger
+import java.util
+import java.util.{Optional => JOptional}
+import scala.collection.JavaConverters.seqAsJavaListConverter
+import scala.collection.mutable.ListBuffer
 import scala.concurrent.duration.DurationInt
 import scala.concurrent.{Await, Future}
-import scala.util.{Failure, Success}
 import scala.language.postfixOps
-import java.util.{Optional => JOptional}
-import java.math.BigInteger
-import java.util
+import scala.util.{Failure, Success}
 
 
 class EthService(val stateView: AccountStateView, val nodeView: CurrentView[AccountHistory, AccountState, AccountWallet, AccountMemoryPool], val networkParams: NetworkParams, val sidechainSettings: SidechainSettings, val sidechainTransactionActorRef: ActorRef) extends RpcService {
 
-  @RpcMethod("eth_getBlockByNumber") def getBlockByNumber(tag: Quantity, rtnTxObj: Boolean): EthBlock.Block = { //TODO: Implement getting block information
-    new EthBlock.Block("0x1", "0x56", "0x57", "0x58", "0x59", "0x0", "0x0", "0x0", "0x0", "0", "0", "0", "0", "0", "0", "1", "3000", "2000", "22", new util.ArrayList[EthBlock.TransactionResult[_]], new util.ArrayList[String], new util.ArrayList[String], "")
-  }
-
-
-  @RpcMethod("eth_call") def call(ethCall: util.LinkedHashMap[String, String], tag: Quantity): Data = { // Executes a new message call immediately without creating a transaction on the block chain
-    // TODO: add read-only evm execution and cleanup here
-    var processor = new EvmMessageProcessor();
-    val msg = new Message(
-      new AddressProposition(Numeric.hexStringToByteArray("0xA83035b4E719827f146C91b119508ea452c77F35")),
-      new AddressProposition(Numeric.hexStringToByteArray(ethCall.get("to"))),
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      BigInteger.ZERO,
-      Numeric.hexStringToByteArray(ethCall.get("data")))
-    if (processor.canProcess(msg, stateView)) {
-      var result = processor.process(msg, stateView)
-      if (!result.isFailed)
-        return new Data(result.asInstanceOf[ExecutionSucceeded].returnData())
+  //function which describes default transaction representation for answer after adding the transaction to a memory pool
+  val defaultTransactionResponseRepresentation: (Transaction => SuccessResponse) = {
+    transaction => TransactionIdDTO(transaction.id)
+  }
+
+  @RpcMethod("eth_getBlockByNumber") def getBlockByNumber(tag: Quantity, hydratedTx: Boolean): EthereumBlock = {
```

Comment:
to avoid code duplication, after retrieving `blockId` you may call `getBlockByHash`